packages/scaffold-config/src/detect.ts

```diff
@@ -64,3 +71,83 @@ export function detect (projectPath: string): DetectFramework {
     bundler: undefined,
   }
 }
+
+/**
+ * Detect the language the current project is using
+ *
+ * If `cypress.config` exists, we derive the language
+ * from the extension.
+ *
+ * IF HAS_CYPRESS_CONFIG
+ *   IF CYPRESS_CONFIG_TS
+ *     HAS TYPESCRIPT
+ *   ELSE
+ *     DOES NOT HAVE TYPESCRIPT
+ *   ELSE IF `typescript` dependency in `package.json` OR `tsconfig.json` in `projectRoot/*`
+ *     HAS TYPESCRIPT
+ *   ELSE
+ *     DOES NOT HAVE TYPESCRIPT
+ *   END
+ * ELSE IF HAS CYPRESS_JSON
+ *   IF cypress/* contains *.ts file
+ *     USE TYPESCRIPT
+ *   ELSE
+ *     DO NOT USE TYPESCRIPT
+ *   END
+ * ELSE IS NEW PROJECT
+ *   IF `typescript` dependency in `package.json` OR `tsconfig.json` in `projectRoot/*`
+ *     HAS TYPESCRIPT
+ *   ELSE
+ *     DOES NOT HAVE TYPESCRIPT
+ *   END
+ * END
+ */
+
+export function detectLanguage (projectRoot: string, pkgJson: PkgJson): 'js' | 'ts' {
+  try {
+    if (fs.existsSync(path.join(projectRoot, 'cypress.config.ts'))) {
+      debug('Detected cypress.config.ts - using TS')
+
+      return 'ts'
+    }
+
+    if (fs.existsSync(path.join(projectRoot, 'cypress.config.js'))) {
+      debug('Detected cypress.config.js - using JS')
+
+      return 'js'
+    }
+  } catch (e) {
+    //
```

Comment:
Worth debug logging this error?

Reply:
Sure, https://github.com/cypress-io/cypress/pull/21220/commits/a8cdfb10f2dea22a41165f15e8dc453def876d44

Also removed superfluous try/catch.